### PR TITLE
Clang and contributing

### PR DESCRIPTION
**Pull request by NinjaDanz3r**
A step towards clanging everything.

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,6 +24,16 @@ A class should only declare one specifier per specifier type.
 All non-private members need to be documented using Doxygen-style comments.
 For functions and methods, parameters are documented first and return type next.
 
+### Clang
+We use clang-format (v 3.9) to enforce our coding standard.
+To format a specific file, use clang-format -i PATH_TO_FILE/FILENAME when in the /src directory.
```
Comment (Chainsawkitten) [inline CONTRIBUTING.md]: Use `` around the command to make it clear where it begins and ends, like `this`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,6 +24,16 @@ A class should only declare one specifier per specifier type.
...
+
+Clang-format for windows can be grabbed from here: 
+http://releases.llvm.org/
+
+The options we use can be found here:
```
Comment (Chainsawkitten) [inline CONTRIBUTING.md]: We don't use all these options, though. A clearer explanation may be: "The options we use are specified in `src/.clang-format`. Explanations of the options can be found here:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -33,27 +43,27 @@ For functions and methods, parameters are documented first and return type next.
...
-        
-        /// Get whether the fish is happy.
-        /**
-         * @return Whether the fish is happy.
-         */
-        bool GetHappy() const;
-        
+    /// Create new fish.
+    /**
+    * @param name The name to give the fish.
```
Comment (Chainsawkitten) [inline CONTRIBUTING.md]: Improper * alignment in doxygen comment. (Check if clang-format allows this or if it changes it back again. If it changes it back again, we may have to use a regex to ignore doxygen comments, or live with improperly formatted doxygen comments).

**Pull request merged**